sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/complexTypeExtractors.scala

```diff
@@ -236,8 +236,6 @@ case class GetArrayItem(
     failOnError: Boolean = SQLConf.get.strictIndexOperator)
   extends BinaryExpression with GetArrayItemUtil with ExpectsInputTypes with ExtractValue {
 
-  def this(child: Expression, ordinal: Expression) = this(child, ordinal, SQLConf.get.ansiEnabled)
```

Comment:
The `SQLConf.get.ansiEnabled` here should be `SQLConf.get.strictIndexOperator`. But it doesn't matter since the code here is not reachable.